src/main/java/com/openshift/restclient/ClientBuilder.java

```diff
@@ -183,7 +189,9 @@ public IClient build() {
 			dispatcher.setMaxRequests(maxRequests);
 			dispatcher.setMaxRequestsPerHost(maxRequestsPerHost);
 
-			OkHttpClient.Builder builder = new OkHttpClient.Builder()
+			OkHttpClient.Builder builder = null;
+			if (!this.sslCertCallbackWithDefaultHostnameVerifier) {
```

Comment:
Also, the default behavior should stay the same ... setting an ISSLCertificateCall should both update the trust store with that callback as well as update the hostname verifier in okhttp3.